utils/graph_optimize.py

```diff
@@ -97,7 +97,109 @@ def merge_bn(symbol, args, auxs, symbol_only=False):
+    it back to mxnet graph.
+    """
+    assert symbol is not None
+    assert base_quant_attrs is not None
+    # currently only support quant_mode = "minmax" and weight per tensor quantization method
+    base_quant_attrs["is_weight"] = "False"
+    base_quant_attrs["is_weight_perchannel"] = "False"
+    base_quant_attrs["quant_mode"] = "minmax"
+
+    data_quant_attrs = base_quant_attrs.copy()
+    weight_quant_attrs = base_quant_attrs.copy()
+    weight_quant_attrs["is_weight"] = "True"
+
+    jgraph = json.loads(symbol.tojson())
+    jnodes = jgraph["nodes"]
+    node_map = {}
+    node_op_map = {}
+    quantized_node_map = {}
+
+
+    for nid, node in enumerate(jnodes):
+        # edges are [which_node, which_output, type(? not sure)]
+        # mx.symbol has an attribute of __getitem__. sym[1] gives the second output
+        children = [node_map[e[0]][e[1]] for e in node["inputs"]]
+        attrs = node.get("attrs", {})
+        node_name = node["name"]
+        op_name = node["op"]
+        if op_name == "null":
+            attrs = dict({k:v for k, v in attrs.items() if k.startswith("__")})
+            assert node_name in out_shape_dict.keys(), "{} Variable is not in shape_dict".format(node_name)
+            if "__shape__" not in attrs.keys():
+                attrs["__shape__"] = out_shape_dict[node_name]
+                attrs["__dtype__"] = 0  # "float32"
+            node_map[nid] = mx.sym.var(node_name, **attrs)
+            node_op_map[nid] = ["Variable"]
+        elif op_name in quantized_op:
+            if op_name in ["Convolution", "FullyConnected", "Deconvolution"]:
+                if len(children) == 2:
+                    datavar, weightvar = children
+                    biasvar = None
+                else:
+                    datavar, weightvar, biasvar = children
+                data_name, weight_name = datavar.name, weightvar.name
+                if data_name in quantized_node_map.keys():
+                    print("{} has attached quantized node".format(data_name))
+                    data_quanted = quantized_node_map[data_name]
+                else:
+                    data_quanted = mx.sym.contrib.Quantization_int8(datavar, **data_quant_attrs, name=data_name)
+                    quantized_node_map[data_name] = data_quanted
+                if weight_name in quantized_node_map.keys():
+                    print("{} has attached quantized node".format(weight_name))
+                    weight_quanted = quantized_node_map[weight_name]
+                else:
+                    weight_quanted = mx.sym.contrib.Quantization_int8(weightvar, **weight_quant_attrs, name=weight_name)
+                    quantized_node_map[weight_name] = weight_quanted
+                print("attach quantize node for {} inputs:{}, {}".format(op_name, data_name, weight_name))
+                quanted_children = [data_quanted, weight_quanted, biasvar]
+            elif op_name in ["Concat", "Pooling", "add_n", "elemwise_add"]:
```

Comment:
Concat has another verison named concat. So as the Elementwise. Please check other ops to make sure all versions are properly supported.

Reply:
it seems like only `concat` has alias `concat or Concat`, `elemwise_add` has no alias